### ISSUE TITLE
Use available indexes when in `CaseTransformFunction`

### ISSUE THREAD

**Issue by mayankshriv** [apache/pinot#7691]
We are seeing various cases where auto generated SQL from visualization tools perform a lot of transforms/filtering using `CASE WHEN` from within aggregation functions. However, if the original predicate in the `where clause` had low selectivity, we end up scanning a lot of data to perform the `CaseTransform` evaluation. It would be good to see if we can utilize available indexes (when possible) for evaluating these as well.

**Comment by atris** [comment]
Let's just promote the use of FILTER clauses instead of Case When, since it will be able to use the relevant index

**Comment by mayankshriv** [comment]
Sure, but that would mean we re-write the query internally, as many a times users may not directly control the query generated for Pinot.

**Comment by atris** [comment]
Yes, good point. Let's rewrite all case queries to use the filter plan instead

**Comment by weixiangsun** [comment]
Are we going to run multiple queries corresponding to each condition inside the CaseTransform? @atris

**Comment by atris** [comment]
No, a single scan with multiple predicate evaluations along the way. Please see:

https://github.com/apache/pinot/pull/7830

**Comment by weixiangsun** [comment]
I see. This is used to reduce the data set for aggregation. My use case is:
select case
when timeStamp IS NULL then now()
else timestamp
end
from meetupRsvp

https://github.com/apache/pinot/issues/7835 is different that there might be no aggregation in my use case. It will generate one column. it will generate different columns inside the resultset.
